kac-api/src/index.ts

```diff
@@ -1,128 +1,64 @@
-import {readFileSync} from "fs";
-import path from "path";
-import fastify from "fastify";
-import {V1ValidatingWebhookConfiguration, V1ObjectMeta} from "@kubernetes/client-node";
-import {createDefaultMonokleValidator, Resource} from "@monokle/validation";
-
-// TODO - some mismatch with types, this is not exactly the type which should be used
-type AdmissionRequest = V1ValidatingWebhookConfiguration & {
-    request?: V1ObjectMeta & {
-      object?: Resource
-    }
-}
-
-type AdmissionResponse = {
-    kind: string,
-    apiVersion: string,
-    response: {
-        uid: string,
-        allowed: boolean,
-        status: {
-            message: string
-        }
-    }
-}
-
-const validator = createDefaultMonokleValidator();
-
-const server = fastify({
-  https: {
-    key: readFileSync(path.join('/run/secrets/tls', 'tls.key')),
-    cert: readFileSync(path.join('/run/secrets/tls', 'tls.crt'))
-  }
-});
-
-server.post("/validate", async (req, res): Promise<AdmissionResponse> => {
-  console.log('request', req.headers, req.body)
-
-  const body = req.body as AdmissionRequest;
+import pino from 'pino';
+import {AnnotationSuppressor, FingerprintSuppressor, MonokleValidator, RemotePluginLoader, SchemaLoader, DisabledFixer, ResourceParser} from '@monokle/validation';
+import {MonoklePolicy, getInformer} from './utils/informer.js';
+import {ValidationServer} from './utils/validation-server.js';
+import {DEFAULT_NAMESPACE, getNamespace} from './utils/helpers.js';
+
+(async() => {
+  const logger = pino({
+    name: 'Monokle',
+    level: 'trace',
+  });
 
-  const response = {
-    kind: body?.kind || '',
-    apiVersion: body?.apiVersion || '',
-    response: {
-      uid: body?.request?.uid || "",
-      allowed: true,
-      status: {
-        message: "OK"
-      }
-    }
+  let currentNamespace: string;
+  try {
+    currentNamespace = getNamespace() || DEFAULT_NAMESPACE;
```

Comment:
Wait, so the way you built it would require the Admission Controller to be deployed in each namespace in order to validate resources? Did I get that right?

If yes, why? Wouldn't an aproach where the operator would be deployed globally work better & be more reliable?